omdata/electrolyte_utils.py

```diff
@@ -0,0 +1,77 @@
+from typing import Any, Dict, List, Set, Tuple
+
+import numpy as np
+
+from rdkit import Chem
+from rdkit.Chem.rdchem import Mol
+
+from pymatgen.core.structure import Molecule
+from pymatgen.io.ase import AseAtomsAdaptor
+from pymatgen.io.babel import BabelMolAdaptor
+
+
+def info_from_smiles(
+    smiles: Dict[str, str] | List[str] | Set[str]
+    ) -> Dict[str, Any]:
+    """
+    Generate Calculate the number of atoms in a molecule from SMILES.
```

Comment:
```suggestion
   Convert SMILES to a variety of formats and also retrieve some metadata.
```

This does not do what it says on the tin. Why do we need SMILES in three different formats? You don't seem to be using the pmg_mol or ase_atoms info anywhere. You can extract charge from the RDKit mol with Chem.GetFormalCharge(mol) and the number of unpaired electrons with (from rdkit.Chem import Descriptors) Descriptors.NumRadicalElectrons(mol). Also, if you do Chem.AddHs(mol) after you make the mol, you can get the number of atoms and heavy atoms as mol.GetNumAtoms() and mol.GetNumHeavyAtoms(), respectively. So I think you might just want to return a dict of rdkit_mols

Reply:
Yep, this is a mess. I'll clean it up.